src/primitives/transaction.cpp

```diff
@@ -209,12 +209,12 @@ std::list<COutPoint> CTransaction::GetOutPoints() const
 
 CAmount CTransaction::GetZerocoinSpent() const
 {
-    if(!IsZerocoinSpend())
+    if(!IsZerocoinSpend() && !IsZerocoinPublicSpend())
```

Comment:
better to unify them in a single method `IsZerocoinSpend` for both and then have two separate for specific scenarios. Otherwise we'll have to check and change every IsZerocoinSpend instance for public spends checks too as it starts with another opcode and the validation flow is basically the same for both.